### DISCUSSION TITLE
Port Forwarding: failed to forward localhost:5500, network error

### DISCUSSION THREAD

**Discussion by callmeclover** [community/community#68324] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

hello! i am trying to run live server, and it wont forward the port 5500. i can't do it manually either, as it shows the error ```Unable to forward localhost:5500. Network error.``` do any of you have a similar issue? a workaround, or fix?

choosing a different port does nothing either, i cant forward any of them.

**Comment by ThiagoCarmona** [comment]
I'm having the same problem

**Comment by MohamedYansane** [comment]
Hi guys! go to the extensions and look for live server and clik to the log icons of your live server then look for use browser preview and unselect it if it's checked and select the use local ip 😉. If it worked please marked it as resolved thanks

**Reply by callmeclover** [reply]
hello, this does not solve my issue, it just changed it to my own ip, which doesn't work. when disabled, it does open a webpage, but it doesn't load, because there is no port forwarded. i would also like to add it does not open a codespace app link, but rather 127.0.0.1 localhost.

**Comment by MohamedYansane** [comment]
try to write manually on your browser after doing what i mentioned above 127.0.0.1:5500

**Reply by callmeclover** [reply]
still nothing, anything else?

**Comment by nottisa** [comment]
Code Spaces seems to be giving me trouble as well. The automatic port forwarding doesn't seem to work, and when I manually try to forward ports it just gives a network error. This seems to happen across several web based repositories.

**Comment by alexec** [comment]
Port forwarding appears to be intermittently broken. I don't know why Github does not know this.

**Comment by MohamedYansane** [comment]
my last card go to vscode settings and look for liver server and in  live server>Settings:Https check if enable is set to false

**Reply by nottisa** [reply]
This only works if you are developing using live server, personally, I'm using Strawberry and Apache. Please stop suggesting things that do not pertain to people's issues.

**Reply by MohamedYansane** [reply]
above u didn't mentionned u r using strawberry and Apache that's why sorry then hope u got a help or find a solution thanks

**Comment by callmeclover** [comment]
An update: I'm sending a ticket to GitHub support.

Hopefully they have something of a fix. If they do, I'll post it here. I'm no masochist, after all.

**Comment by Findeton** [comment]
Port forwarding seems to be broken on codespaces. Our current fix is to use a proxy to connect to the containers but this is prone to errors. Please fix it, we've been battling with this for days.

**Comment by GLStephen** [comment]
We are seeing this issue also. Also reported here: https://github.com/orgs/community/discussions/68854

**Comment by osortega** [comment]
Hello all!
I'm sorry for all the friction you've been having.

I've read the reports here and it seems that most have had a `Network Error` which comes from within the `axios` library that is used to do API requests.
I suspect this issue would surface in the network tab of the developer tools and indicates that something is blocking the request in the browser. It could be a local firewall or maybe a VPN.

Do you mind confirming if this is the case?

**Reply by callmeclover-alt** [reply]
I think this may be my case, considering I am using a school distributed computer. Sucks, though.

**Comment by alexec** [comment]
- I don't have any browser plugins.
- This issue seems to have started happening about 1 week ago.
- It seems to be intermittent for me.
- Occurs in both Safari and Chrome.

**Comment by Findeton** [comment]
@osortega I think it must be something more fundamental. I've tried different browsers, different computers, Windows, Linux, Mac... Surely not all of them would have a firewall, also given that it was working a week ago. And just like everyone else this has happened since about a week ago. I'd tend to think this is an error from the codespaces side.

**Comment by alexec** [comment]
Another point I'd add. When it does work (about 1 in 20 times), Codespaces automatically detects the port is opened and asks me if I want to forward it.

It seems very unlikely to be a browser issue for me.

**Comment by osortega** [comment]
@alexec @Findeton are you also seeing the `Network Error` issue posted in this discussion?
Could you also share a codespace name that showed this behavior so that I could take a closer look?

**Reply by GLStephen** [reply]
We are also seeing this: "ominous engine"

**Reply by osortega** [reply]
@GLStephen I looked at your codespace but I don't see the error pattern described above.
Is your scenario different? Do you mind explaining a bit what are you seeing?

**Reply by GLStephen** [reply]
I'm seeing nearly the exact same behavior as all others described here. Port forwarding from the desktop client stopped working ~ last Friday regardless of visibility. Loading the codespace in a browser works if we set the visibility of the port to public.

**Reply by mikebronner** [reply]
I have been able to get all but postgres port 5432 to forward correctly when running in the browser. Has anyone else tried forward postgres through the browser and access it from a local database client?

**Comment by alexec** [comment]
Sure:

https://supreme-rotary-phone-7v57jg4g9x2qxq.github.dev/

**Reply by osortega** [reply]
@alexec I looked at this codespace and it seems that there were 3 port forwarding failures and all of them related to `Network Error` from `axios` library.
When this happen, do you mind checking the network tab of the browser in developer tools to find the failed request and see the details on why that was the case?
That would give us more information on why the browser is blocking the request.

**Comment by GLStephen** [comment]
@osortega is there a description/flowchart of exactly what is involved in making that port forwarding work?

**Comment by nottisa** [comment, marked as answer]
@osortega @callmeclover It seems to be an issue with the firewall in my case. My network is blocking one of the port forwarding domains labeling it as "proxy avoidance". Check this in your browsers network logs. In my case it only seems to be labeling GET/POST requests in this manner.
![Uploading Screenshot 2023-10-04 120549.png…]()

**Reply by callmeclover-alt** [reply]
I think this may be my case, considering I am using a school distributed computer. Sucks, though.

**Comment by alexec** [comment]
It was not working 2h ago. It is working now. I tried both with and without VPN. Both work. I get the automatic pop-up:

![image](https://github.com/community/community/assets/1142830/997edbda-40f2-48fa-aea6-e673c7517085)

**Comment by AnimMouse** [comment]
Unblock this domain in your firewall: `tunnels.api.visualstudio.com`

**Comment by chmod777john** [comment]
About one week ago, everything works well. And today I meet the same error.
```
Uable to forward localhost:3000. Network Error.
We noticed that you were unable to forward a port due to an issue with your network. Our port forwarding infrastructure has been upgraded and there might be additional actions for you to take to in order to avoid disruption.
```
![31D8C60PYBTWMOW~M)KC 04](https://github.com/community/community/assets/146207719/f4b23269-d7be-4694-914f-b58e4f51d82f)

I have tried on two of my windows. And one of my friend also tried on his. None of the 3 windows works.

However, I have tested on this, and it works well.

https://github.com/fcwu/docker-ubuntu-vnc-desktop

I open this project in a codespace, and then `docker run -p 6080:80 -p 5900:5900 -v /dev/shm:/dev/shm dorowu/ubuntu-desktop-lxde-vnc` to start the remote ubuntu desktop.

From the remote desktop, I installed VSCode and extensions, and open a codespace with this "remote desktop VSCode", everything works well. The port can be forwarded.

So I don't think it is a firewall issue. I believe the Github team or VSCode team have done something that lead to this bug. It may be temporary or not.

By the way, I live in China and I use a VPN. (We cannot connect to Github without a VPN.) I'm not sure if that matters. The 3-Windows-tests above all use a VPN tool called clash. Here is the link https://github.com/MetaCubeX/mihomo/tree/Alpha

**Reply by chmod777john** [reply]
The lowercase "windows" above all represent the Windows OS.
